src/components/pages/contact/contact-form.tsx

```diff
@@ -56,101 +73,62 @@ export const ContactForm: React.FC = () => {
-      onSubmit={handleSubmit(onSubmit)}
-    >
-      <Grid nested>
-        {/*First Name*/}
-        <GridItem xs={12} md={6}>
-          <InputField
-            required={true}
-            inputRef={register('name', {
-              required: t<string>('contact.error.name'),
-            })}
-            error={errors.name}
-            name="name"
-            label={t('contact.name')}
-          />
-        </GridItem>
-        <GridItem xs={12} md={6}>
-          <InputField
-            required={true}
-            inputRef={register('email', {
-              required: t<string>('contact.error.email'),
-              pattern: {
-                value: SIMPLE_EMAIL_PATTERN,
-                message: t<string>('contact.error.email-unknown'),
-              },
-            })}
-            error={errors.email}
-            name="email"
-            label={t<string>('contact.email')}
-          />
-        </GridItem>
-        <GridItem>
-          <InputField
-            required={true}
-            inputRef={register('message', {
-              required: t<string>('contact.error.message'),
-            })}
-            error={errors.message}
-            name="message"
-            label={t('contact.message')}
-            type={'text-area'}
-          />
-        </GridItem>
-        <GridItem>
-          <StyledCaptionText>
-            <Sup>*</Sup> {t('contact.mandatory-field')}
-          </StyledCaptionText>
-        </GridItem>
-        <GridItem>
-          {requestStatus === 'pending' && (
-            <SendButton type="submit">
-              <ButtonText>{t('contact.action.send')}</ButtonText>{' '}
-              <RightArrowIcon />
-            </SendButton>
-          )}
-          {(errors.name || errors.email || errors.message) && (
-            <ErrorMessageSend>{t('contact.action.missing')}</ErrorMessageSend>
-          )}
-          {requestStatus === 'success' && (
-            <SentButton type="button">
-              <ButtonText>{t('contact.action.sent')}</ButtonText>{' '}
-              <CheckmarkIcon />
-            </SentButton>
-          )}
-          {requestStatus === 'error' && (
-            <>
-              <Trans i18nKey="contact.action.again-text">
-                Leider gab es einen Fehler. Bitte versuche es noch einmal.
-                Klappt das nicht, schicke deine Nachricht bitte direkt an
-                <Link to="mailto:beep@satellytes.com">beep@satellytes.com</Link>
-              </Trans>
-              <SendButton type="submit">
-                <ButtonText>{t('contact.action.again')}</ButtonText>{' '}
-                <RightArrowIcon />
-              </SendButton>
-            </>
-          )}
-        </GridItem>
-      </Grid>
-    </form>
+    <>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <StyledGrid nested>
+          <GridItem xs={12} md={6}>
+            <TextInput
+              name={'name'}
+              label={t('contact.name')}
```

Comment:
Did you create the contact form again or did you adapt the legacy one? I thought you build from scratch (for fun) ? This looks like a legacy adaption. I would like to get rid of the grid and we should rather define a local set of divs with the appropriate css grid. We don't need a shared grid component. This looks super noisy and I think there is a lot of refactor potential. Maybe you can just leave the contact form as is and instead create a second contact form we can improve slowly (via the storybook) and later just replace it on the contact page.

It's not wrong what you are doing here. If you feel comfortable we can move forward with that approach too.

Reply:
I build everything from scratch except the grid and the handleSubmit method. But I agree with you, it would be nice if we get rid of the gird